### PR TITLE
Adding daily calories goal, and schema to show on homepage.

### PR DESCRIPTION
**Pull request by Gbaker97**
User Profile:
- Changes to add in a exercise aim (fitness, weight loss etc), and a daily calories goal.
- Adding these values to exercise class in activity tracking backend.
Analytics backend:
- Set-up for schema to retrieve the daily calories goal.
Homepage:
- Retrieving goal via graphql schema.
- Changes to frontend to use goal on radial chart.
image
image

### CODE REVIEW

File: activity-tracking/routes/goals.js [hunk 1]
```diff
@@ -38,9 +40,9 @@ router.post('/add', async (req, res) => {
 // PUT: Update the user's goal by username (User should only have one goal)
 router.put('/update/:username', async (req, res) => {
     try {
-      const { username, exerciseType, goalType, goalUnit, goalValue } = req.body;
+      const { username, exerciseType, goalType, goalUnit, goalValue, caloriesGoal, goalAim } = req.body;
```
Comment (ursperry) [inline activity-tracking/routes/goals.js:43]: What is the goal aim doing? storing the kcal aimed for right?

File: activity-tracking/routes/goals.js [hunk 2]
```diff
@@ -18,14 +18,16 @@ router.post('/add', async (req, res) => {
...
+    const { username, exerciseType, goalType, goalUnit, goalValue, caloriesGoal, goalAim } = req.body;
 
     const newGoal = new Goal({
       username,
       exerciseType,
       goalType,
       goalUnit,
-      goalValue: Number(goalValue)
+      goalValue: Number(goalValue),
+      caloriesGoal: Number(caloriesGoal),
```
Comment (ursperry) [inline activity-tracking/routes/goals.js:29]: what is the difference between goalValue and caloriesGoal please?

File: analytics/app.py [hunk 3]
```diff
@@ -126,6 +126,11 @@ def resolve_home_page(*_, name=None):
     def resolve_daily_calories(*_, name=None, today_date=None):
         print("Resolving the daily calorie info")
         return resolve_query(func=daily_calories, username=name, today_date_str=today_date)
+
+    @query.field("caloriesGoal")
```
Comment (ursperry) [inline analytics/app.py:130]: is this pulling from the new GraphQL schema/kcal?

File: frontend/src/components/homepage.js [hunk 4]
```diff
@@ -4,9 +4,9 @@ import { Link } from 'react-router-dom';
 import 'bootstrap/dist/css/bootstrap.min.css';
 import IconButton from '@mui/material/IconButton';
 import { useQuery } from '@apollo/client';
-import { LAST_EXERCISE_QUERY, CALORIES_QUERY } from './queries/graphql';
+import { LAST_EXERCISE_QUERY, CALORIES_QUERY, CALORIES_GOAL_QUERY } from './queries/graphql';
 import { Mood, MoodBad, SentimentSatisfied, SentimentVeryDissatisfied } from '@mui/icons-material'; 
-import { BarChart, Bar, RadialBarChart, RadialBar, PolarAngleAxis, ResponsiveContainer, XAxis, YAxis, Tooltip, Cell} from 'recharts';
+import { RadialBarChart, RadialBar, PolarAngleAxis, ResponsiveContainer, Tooltip, Cell} from 'recharts';
```
Comment (ursperry) [inline frontend/src/components/homepage.js:9]: is this the diagram import, cool!

**Review by ursperry — COMMENTED** [review]
that looks great Georgie!! thank you

File: activity-tracking/routes/goals.js (hunk 2 above)
Comment (Gbaker97) [reply activity-tracking/routes/goals.js:29]: goalValue is the amount associated with the weekly exercise goal. Calories goal, is a value for the daily calories goal that the user can set as well.

File: activity-tracking/routes/goals.js (hunk 1 above)
Comment (Gbaker97) [reply activity-tracking/routes/goals.js:43]: goalAim is a property of the Goal class, which stores the value that the user selects as their aim, for example, fitness, weight loss etc.

**Pull request merged**